examples/chat/client/cpp/chat.cc

```diff
@@ -113,64 +112,44 @@ void SendLoop(Chat::Stub* stub, const RoomId& room_id, const std::string& user_h
+void Chat(Chat::Stub* stub, const std::string& user_handle) {
   // TODO(#746): make both loops notice immediately when done is true.
   auto done = std::make_shared<Safe<bool>>(false);
 
   // Start a separate thread for incoming messages.
-  std::thread listener([stub, room_id, &user_handle, done] {
-    LOG(INFO) << "New thread for incoming messages in room ID " << absl::Base64Escape(room_id);
-    ListenLoop(stub, room_id, user_handle, done);
+  std::thread listener([stub, &user_handle, done] {
+    LOG(INFO) << "New thread for incoming messages in room";
+    ListenLoop(stub, user_handle, done);
     LOG(INFO) << "Incoming message thread done";
   });
   listener.detach();
 
   std::cout << "\n\n\n";
-  SendLoop(stub, room_id, user_handle, done);
+  SendLoop(stub, user_handle, done);
 }
 
-// RAII class to handle creation/destruction of a chat room.
-class Room {
- public:
-  // Caller should ensure stub outlives this object.
-  Room(Chat::Stub* stub) : stub_(stub) {
-    oak::NonceGenerator<64> generator;
-    grpc::ClientContext context;
-    CreateRoomRequest req;
-    auto room_id = generator.NextNonce();
-    auto room_id_string = std::string(room_id.begin(), room_id.end());
-    req_.set_room_id(room_id_string);
-    auto admin_token = generator.NextNonce();
-    req_.set_admin_token(std::string(admin_token.begin(), admin_token.end()));
-    google::protobuf::Empty rsp;
-    grpc::Status status = stub_->CreateRoom(&context, req_, &rsp);
-    if (!status.ok()) {
-      LOG(FATAL) << "Could not CreateRoom('" << absl::Base64Escape(room_id_string)
-                 << "'): " << oak::status_code_to_string(status.error_code()) << ": "
-                 << status.error_message();
-    }
-  }
-  ~Room() {
-    LOG(INFO) << "Destroying room";
-    grpc::ClientContext context;
-    DestroyRoomRequest req;
-    req.set_admin_token(req_.admin_token());
-    google::protobuf::Empty rsp;
-    grpc::Status status = stub_->DestroyRoom(&context, req, &rsp);
-    if (!status.ok()) {
-      LOG(WARNING) << "Could not DestroyRoom(): " << oak::status_code_to_string(status.error_code())
-                   << ": " << status.error_message();
-    }
+// Create a gRPC stub for an application, with the provided room access token, which will be used as
+// confidentiality label for any messages sent, and also to authenticate to the application in order
+// to read messages sent by other clients.
+std::unique_ptr<Chat::Stub> create_stub(std::string address, std::string ca_cert,
+                                        std::string room_access_token) {
+  // TODO(#1357): Use a public / secret key pair instead of bearer token credentials. In fact,
+  // currently not even bearer token credentials can be used, because the privilege is not correctly
+  // assigned to the gRPC server node, and any responses would never reach the client, so we use a
+  // public label as placeholder.
+  oak::label::Label label = oak::PublicUntrustedLabel();
```

Comment:
nit: This contradicts the title and description for the PR. perhaps update these to make clear this is a preparatory PR that would help support meaningful labels once the gRPC server node can handle the privilege correctly.

Reply:
Clarified in the description and title of the PR, I also scattered TODOs around so hopefully it will not stay like this for long.